qiskit/opflow/gradients/circuit_qfis/overlap_block_diag.py

```diff
@@ -167,16 +167,15 @@ def get_parameter_expression(circuit, param):
                     block[i][j] = psi_gen_ij - cross_term
 
                     # pylint: disable=unidiomatic-typecheck
-                    if type(param_expr_i) == ParameterExpression:
```

Comment:
I think this is an equality check on purpose, since the derivative of a `Parameter` is always 1 but the derivative of a `ParameterExpression` might not be. Thus, to skip `Parameter`s we don't do an `isinstance` check

Reply:
ok, makes sense (since you also disabled unidiomatic-typecheck, it is on purpose)